app/services/analytics_events.rb

```diff
@@ -248,4 +248,65 @@ def proofing_address_result_missing
   def proofing_document_result_missing
     track_event('Proofing Document Result Missing')
   end
+
+  # @identity.idp.event_name Return to SP: Failed to proof
+  # Tracks when a service provide fails to proof.
+  # @param [String] redirect_url the url of the service provider
+  def return_to_sp_failure_to_proof(url, **location_params)
+    track_event(
+      'Return to SP: Failed to proof',
+      redirect_url: url,
+      **location_params,
+    )
+  end
+
+  # @identity.idp.event_name Rules of Use Visited
+  # Tracks when rules of use is visited
+  def rules_of_use_visit
+    track_event('Rules of Use Visited')
+  end
+
+  # @identity.idp.event_name Rules of Use Submitted
+  # Tracks when rules of use is submitted with a success or failure
+  # @param [Hash] result_hash hash containing the result of accepting the rules of use
+  def rules_of_use_submitted(result_hash, **extra)
+    track_event(
+      'Rules of Use Submitted',
+      result_hash,
+      **extra,
+    )
```

Comment:
can can we document what's inside `result_hash`? Usually there are `sucess:` `errors:` and some other extra attributes